src/map/status.cpp

```diff
@@ -13495,16 +13455,21 @@ int status_change_end(struct block_list* bl, enum sc_type type, int tid)
 				pc_delservantball( *sd, sd->servantball );
 			}
 			break;
-		case SC_CHARGINGPIERCE:
-			status_change_end(bl, SC_CHARGINGPIERCE_COUNT);
-			break;
 		case SC_ABYSSFORCEWEAPON:
 			if( sd ){
 				pc_delabyssball( *sd, sd->abyssball );
 			}
 			break;
 	}
 
+	// End statuses found in the Cancel list.
+	if (!scdb->cancel.empty()) {
+		for (const auto &it : scdb->cancel) {
+			if (it == type)
```

Comment:
This if looks weird.
scdb is already the database link for this specific SC.
So the list should be all SCs that should be cancelled and those should most likely be other SCs.
So maybe you simply inverted the check, because you wanted to prevent infinity loops?
But even infinity loops cant be a thing, because sce is already set to NULL at this point.